### PR TITLE
Added PrivateLink service documentation:

### PR DESCRIPTION
**Pull request by meker12**
## Purpose of this pull request

Added information about [PrivateLink service availability](https://devdocs.magento.com/cloud/project/privatelink-service.html) to establish secure, private connections  between Magento Cloud Pro Staging and Production environments and external services.  Information includes
- PrivateLink overview, available support, and features 
- Prerequisites for PrivateLink enablement on Magento Cloud projects
- Enablement process workflow
- Test PrivateLink connection with telnet
- Provided AWS and Azure documentation links for PrivateLink overview,
concepts, and detailed configuration instructions

## Affected DevDocs pages

- https://devdocs.magento.com/cloud/bk-cloud.html
- https://devdocs.magento.com/cloud/project/privatelink-service.html

### CODE REVIEW

File: src/cloud/project/privatelink-service.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+-  A Cloud account (AWS or Azure) in the same region as the {{site.data.var.ece}} instance
+-  A VPC in the customer environment that hosts the services to connect via PrivateLink
+-  For bidirectional PrivateLink connections, your must configure the PrivateLink service in your VPC environment before requesting Magento PrivateLink enablement. See [Customer set up for bidirectional PrivateLink communication][#customer-set-up-for-bidirectional-privateLink-communication].
+-  Gather the following data required for PrivateLink enablement:
+
+   -  **Cloud account number** (AWS or Azure)–Must be in the same region as the Magento Cloud instance
+   -  **Cloud region**–Provide the Cloud region where the account is hosted for verification purposes
+   -  **Services and communication ports**–Magento must open ports to enable service communication between VPCs, for example _Webserver, HTTP port 80_, _SFTP port 2222_
+   -  **Magento Cloud Project ID**
```
Comment (hguthrie) [inline src/cloud/project/privatelink-service.md:60]: The only one without a statement. Maybe can identify where to get the ID just so there is something there.

File: src/cloud/project/privatelink-service.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+   -  **Connection Type**–Specify unidirectional or bidirectional for connection type
```
Comment (hguthrie) [inline src/cloud/project/privatelink-service.md:61]:
```suggestion
   -  **Connection Type**–Specify unidirectional or  bidirectional for connection type
```

File: src/cloud/project/privatelink-service.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+   -  **Service Name**–Service name for the VPC service endpoint that Magento must connect to. Required only to enable bidirectional PrivateLink connections.
```
Comment (hguthrie) [inline src/cloud/project/privatelink-service.md:62]:
```suggestion
   -  **Service Name**–Service name for the VPC service endpoint. Magento requires this only to enable bidirectional PrivateLink connections.
```

File: src/cloud/project/privatelink-service.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+The customer VPC must have a the following resources available to support bidirectional PrivateLink connections:
+
+-  A Network Load Balancer
+-  An endpoint service configuration that enables access to a service from your VPC environment
+-  An [interface endpoint][] (AWS) or [private endpoint][] (Azure) that allows Magento to connect to endpoint services hosted in your VPC
+
+If these resources are not available in the customer VPC, you must sign into your Cloud platform account to add the configuration.
+
+-  Amazon VPC console– ```https://console.aws.amazon.com/vpc/```
```
Comment (hguthrie) [inline src/cloud/project/privatelink-service.md:107]: Not sure why code block ticks are used here. Best to use single ticks unless they are posing a strange problem.

File: src/cloud/project/privatelink-service.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+-  Support for unidirectional or bidirectional communication between Magento and Customer VPCs.
+-  Service enablement–
+   -  Open communication on the necessary ports in the Magento Cloud environment
+   -  Establish the initial connection between the customer and Magento VPC
+   -  Troubleshoot connection issues during enablement
+
+## Limitations
+
+-  Support for PrivateLink is available on Pro plan Production and Staging environments only. It is not available on local or integration environments, or on Starter plan projects.
+-  You cannot establish SSH connections using PrivateLink. Due to infrastructure requirements, Magento cannot open Port 22 for communication between VPCs.
```
[the remark this answers is not in the document]
Comment (hguthrie) [reply src/cloud/project/privatelink-service.md:25]: This is in our style guide, which was why I mentioned it. Ultimately it is writer discretion.

File: src/cloud/project/privatelink-service.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+-  An [interface endpoint][] (AWS) or [private endpoint][] (Azure) that allows Magento to connect to endpoint services hosted in your VPC
+
+If these resources are not available in the customer VPC, you must sign into your Cloud platform account to add the configuration.
+
+-  Amazon VPC console– `https://console.aws.amazon.com/vpc/`
+-  Azure portal– `https://portal.azure.com`
+
+See your Cloud platform documentation for PrivateLink set up instructions:
+
+-  **AWS PrivateLink  documentation**
```
Comment (hguthrie) [inline src/cloud/project/privatelink-service.md:112]: Noticed extra space

```suggestion
-  **AWS PrivateLink documentation**
```

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Comment by meker12** [comment]
running tests

**Pull request merged**